app/src/main/java/com/example/footprnt/Map/MapFragment.java

```diff
@@ -441,13 +453,20 @@ public void onMenuItemClick(View view, int position) {
                                         .withTransparency(0.6f);
                                 mMapRipple.startRippleMapAnimation();      //in onMapReadyCallBack
                                 Intent i = new Intent(getActivity(), FeedActivity.class);
-                                i.putExtra("latitude", latLng.latitude);
-                                i.putExtra("longitude", latLng.longitude);
+                                i.putExtra("latitude", mTappedLocation.latitude);
+                                i.putExtra("longitude", mTappedLocation.longitude);
                                 startActivity(i);
                             }
-                            if (MapConstants.menuItems[position] == MapConstants.DISCOVER){
+                            if (MapConstants.menuItems[position] == MapConstants.DISCOVER) {
                                 //TODO
                             }
+                            if (MapConstants.menuItems[position] == MapConstants.CURRENT) {
```

Comment:
Small optimization but you can do
  if () {
   ...
  } else if () {
    ...
instead of doing only IFs